aws/s3-cdn/s3-cdn.tf

```diff
@@ -512,6 +549,86 @@ module "dns" {
   target_zone_id   = aws_cloudfront_distribution.default.hosted_zone_id
 }
 
+resource "aws_iam_role" "hsts" {
+  count = var.create && var.enable_security_headers ? 1 : 0
+  name  = join(var.delimiter, [local.module_prefix, "hsts", "header"])
+
+  description        = "${var.desc_prefix} Lambda role for s3 cdn HSTS header attachment"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": [
+          "edgelambda.amazonaws.com",
+          "lambda.amazonaws.com"
+        ]
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+
+  tags = local.tags
+}
+
+resource "aws_iam_role_policy" "hsts" {
+  count = var.create && var.enable_security_headers ? 1 : 0
+  name  = join(var.delimiter, [local.module_prefix, "hsts", "header"])
+  role  = aws_iam_role.hsts[0].id
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": "arn:aws:logs:*:*:*"
+    }
+  ]
+}
+EOF
+}
+
+data "archive_file" "hsts" {
+  type        = "zip"
+  output_path = "./dist/hsts.zip"
+  source {
+    content  = var.nodejs_security_header_code
+    filename = "index.js"
+  }
+}
+
+resource "aws_lambda_function" "hsts" {
+  count         = var.create && var.enable_security_headers ? 1 : 0
+  filename      = data.archive_file.hsts.output_path
+  function_name = join(var.delimiter, [local.module_prefix, "hsts"])
+  role          = aws_iam_role.hsts[0].arn
+  handler       = "index.handler"
+  description   = "Blueprint for modifying CloudFront response header implemented in NodeJS."
+
+  source_code_hash = data.archive_file.hsts.output_base64sha256
+
+  runtime     = "nodejs12.x"
+  timeout     = 1
+  memory_size = 128
+
+  tags = local.tags
+
+  tracing_config {
```

Comment:
- [ ]  Determine if X-Ray tracing/logging should be enabled here or in CloudFront so tools like Datadog/AWS can get tracing/logging info

Reply:
Verified logs not provided when set to Active as lambda is deployed to CDN Edge. Set back to PassThrough